Fix group translation strings

NoRi2909 opened the pull request:
Repair messages for Group sensors have broken grammar by mismatching singular and plural.

![image](https://github.com/user-attachments/assets/1d6e5018-26c7-4667-ad62-fa51b204d7ac)

Also it is "unit of measurement" not "measurements" so that needs to be fixed in several occurrences.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Fix all those strings by using "… of measurement" for the unit(s).
Match singular or plural of the nouns and verbs of each message.

The "units of measurement" messages use plural as there can be different ones as long as they can be converted to the resulting unit of the group sensor.

The device class and state class need to be identical for the sources and the group so here the use of singular is correct as we're referring to one class, only.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #129054 
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

NoRi2909 committed:
Group: Fix broken grammar of four issue strings

Repair messages for Group sensor have broken grammar by mismatching singular and plural.

Also it is "unit of measurement" not "measurements" so that needs to be fixed in several occurrences.

joostlek commented on homeassistant/components/group/strings.json:
```diff
@@ -283,20 +283,20 @@
+      "title": "Device class is not correct",
+      "description": "Device class `{device_classes}` on source entities `{source_entities}` needs to be same for sensor group `{entity_id}`.\n\nPlease correct the device classes on the source entities and reload the group sensor to fix this issue."
     },
     "state_classes_not_matching": {
-      "title": "State classes is not correct",
-      "description": "State classes `{state_classes}` on source entities `{source_entities}` needs to be same for sensor group `{entity_id}`.\n\nPlease correct the state classes on the source entities and reload the group sensor to fix this issue."
+      "title": "State class is not correct",
```
I believe `device_classes` and `state_classes` are a list of values seperated by comma's, so the sentence would be
> Device class `measurement, total, total_increasing` on ...

Does that sound logical?

NoRi2909 replied:
Right, if they differ this will be a list, obviously.
Then the "needs" is wrong and we make everything plural:

"XYZ classes … need to be identical" instead of "… needs to be same" (which is bad grammar anyhow)

NoRi2909 committed:
Fix the class mismatch messages by changing the verb to plural

NoRi2909 committed:
Change "Add Group" to "Create Group" for consistency

Following the HA guidelines everything that is created by the user should use "Create …" not "Add …".

Fixing that for consistency with all other Helper types, too.

frenck reviewed (approved):
Nice! Thanks, @NoRi2909 👍 

../Frenck
